akka-http/src/main/scala/akka/http/scaladsl/server/RequestContextImpl.scala

```diff
@@ -109,7 +113,8 @@ private[http] class RequestContextImpl(
   /** Attempts recovering from the special case when non-2xx response is sent, yet content negotiation was unable to find a match. */
   private def attemptRecoveryFromUnacceptableResponseContentTypeException(trm: ToResponseMarshallable, supported: Set[ContentNegotiator.Alternative]): Future[RouteResult] =
     trm.value match {
-      case (status: StatusCode, value) if !status.isSuccess ⇒ this.withAcceptAll.complete(trm) // retry giving up content negotiation
```

Comment:
Yeah, the retry with accept all is pretty "tagged on" here, looking forward to your cleanup of that then - thanks in advance 👍